controllers/accounts_test.go

```diff
@@ -259,6 +269,183 @@ func (suite *AccountsTestSuite) TestAccountDeletionEndpointEnabled() {
+	})
+	finalizeReq.Header.Set("Authorization", "Bearer "+token)
+	finalizeResp := util.ExecuteTestRequest(finalizeReq, suite.router)
+	suite.Equal(http.StatusBadRequest, finalizeResp.Code)
+	util.AssertErrorResponseCode(suite.T(), finalizeResp, util.ErrBadTOTPCode.Code)
+
+	// Test finalizing TOTP setup with valid code
+	validCode, err := totp.GenerateCode(key.Secret(), time.Now().UTC())
+	suite.Require().NoError(err)
+
+	finalizeReq = util.CreateJSONTestRequest("/v2/accounts/2fa/totp/finalize", controllers.TwoFAFinalizeRequest{
+		Code: validCode,
+	})
+	finalizeReq.Header.Set("Authorization", "Bearer "+token)
+	finalizeResp = util.ExecuteTestRequest(finalizeReq, suite.router)
+	suite.Equal(http.StatusOK, finalizeResp.Code)
+
+	var finalizeParsedResp controllers.TwoFAFinalizeResponse
+	util.DecodeJSONTestResponse(suite.T(), finalizeResp.Body, &finalizeParsedResp)
+	suite.Require().NotNil(finalizeParsedResp.RecoveryKey)
+	suite.NotEmpty(*finalizeParsedResp.RecoveryKey)
+
+	// Verify 2FA is now enabled
+	updatedAccount, err := suite.ds.GetAccount(nil, account.Email)
+	suite.Require().NoError(err)
+	suite.True(updatedAccount.IsTwoFAEnabled())
+	suite.NotNil(updatedAccount.RecoveryKeyHash)
+
+	// Test initializing TOTP when it's already enabled
+	initReq = util.CreateJSONTestRequest("/v2/accounts/2fa/totp/init", controllers.TwoFAInitRequest{
+		GenerateQR: true,
+	})
+	initReq.Header.Set("Authorization", "Bearer "+token)
+	initResp = util.ExecuteTestRequest(initReq, suite.router)
+	suite.Equal(http.StatusBadRequest, initResp.Code)
+	util.AssertErrorResponseCode(suite.T(), initResp, util.ErrTOTPAlreadyEnabled.Code)
+}
+
+func (suite *AccountsTestSuite) TestDisableTOTP() {
+	token, account := suite.createAuthSession()
+
+	// Enable 2FA first
+	err := suite.ds.SetTOTPSetting(account.ID, true)
+	suite.Require().NoError(err)
+
+	// Generate and store a recovery key
+	recoveryKey := "test-recovery-key-12345"
+	err = suite.ds.SetRecoveryKey(account.ID, &recoveryKey)
+	suite.Require().NoError(err)
+
+	// Verify timestamps are set
+	details, err := suite.ds.GetTwoFADetails(account.ID)
+	suite.Require().NoError(err)
+	suite.NotNil(details.TOTPEnabledAt)
+	suite.NotNil(details.RecoveryKeyCreatedAt)
+
+	// Test disabling TOTP
+	disableReq := httptest.NewRequest("DELETE", "/v2/accounts/2fa/totp", nil)
+	disableReq.Header.Set("Authorization", "Bearer "+token)
+	disableResp := util.ExecuteTestRequest(disableReq, suite.router)
+	suite.Equal(http.StatusNoContent, disableResp.Code)
+
+	// Verify 2FA is now disabled
+	updatedAccount, err := suite.ds.GetAccount(nil, account.Email)
+	suite.Require().NoError(err)
+	suite.False(updatedAccount.IsTwoFAEnabled())
+	suite.Nil(updatedAccount.RecoveryKeyHash)
+
+	// Verify timestamps are cleared
+	details, err = suite.ds.GetTwoFADetails(account.ID)
+	suite.Require().NoError(err)
+	suite.Nil(details.TOTPEnabledAt)
+	suite.Nil(details.RecoveryKeyCreatedAt)
+}
+
+func (suite *AccountsTestSuite) TestRecoveryKeyEndpoints() {
```

Comment:
Should we test that we can call the regenerate endpoint even when a recovery is not present (e.g. immediately after it's been deleted)?